pandas/core/reshape/pivot.py

```diff
@@ -455,17 +455,23 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
 
     from pandas import DataFrame
     df = DataFrame(data, index=common_idx)
+    common_cols_idx = df.columns
+
     if values is None:
         df['__dummy__'] = 0
         kwargs = {'aggfunc': len, 'fill_value': 0}
     else:
         df['__dummy__'] = values
         kwargs = {'aggfunc': aggfunc}
 
-    table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
+    table = df.pivot_table(['__dummy__'], index=rownames, columns=colnames,
                            margins=margins, margins_name=margins_name,
                            dropna=dropna, **kwargs)
 
+    if not table.empty:
+        added_cols_idx = df.columns.difference(common_cols_idx).values[0]
```

Comment:
is this guaranteed to be len 1?

use
```
added_cols_index = list(df.columns.difference(common_cols_idx))[0]
```

Reply:
Yes. Guaranteed length 1 since __dummy__ is an added column not in original df (if block above the pivot_table call. Will update this line as required.